frigate/detectors/plugins/vim3.py

```diff
@@ -0,0 +1,79 @@
+import logging
+import numpy as np
+
+from frigate.detectors.detection_api import DetectionApi
+from frigate.detectors.detector_config import BaseDetectorConfig
+from typing import Literal
+from pydantic import Extra, Field
+
+try:
+    from tflite_runtime.interpreter import Interpreter, load_delegate
+except ModuleNotFoundError:
+    from tensorflow.lite.python.interpreter import Interpreter, load_delegate
+
+
+logger = logging.getLogger(__name__)
+
+DETECTOR_KEY = "vim3"
+
+
+class vim3DetectorConfig(BaseDetectorConfig):
+    type: Literal[DETECTOR_KEY]
+    device: str = Field(default=None, title="Device Type")
+
+
+class vim3Tfl(DetectionApi):
+    type_key = DETECTOR_KEY
+
+    def __init__(self, detector_config: vim3DetectorConfig):
+        device_config = {"device": "usb"}
+        if detector_config.device is not None:
+            device_config = {"device": detector_config.device}
+
+        edge_tpu_delegate = None
+
+        try:
+            logger.info(f"Attempting to register VIM3 TPU")
+            edge_tpu_delegate = load_delegate("libvx_delegate.so")
+            logger.info("TPU found")
+            self.interpreter = Interpreter(
+                model_path=detector_config.model.path or "/cpu_model.tflite",
```

Comment:
This doesn't seem relevant, if a user requests this it should be running on the TPU otherwise it should fail, I think the CPU model should be left to the existing one.

Reply:
This is deliberate - the Verisilicon TPU runs the CPU model (as the other model is Coral specific)